test/upgrade/test.sh

```diff
@@ -0,0 +1,341 @@
+  echo "Create new schemas for new shards"
+  applySchemaWithRetry create_commerce_seq.sql commerce
+  sleep 4
+  vtctlclient ApplyVSchema -vschema="$(cat vschema_commerce_seq.json)" commerce
+  sleep 4
+  vtctlclient ApplyVSchema -vschema="$(cat vschema_customer_sharded.json)" customer
+  sleep 4
+  applySchemaWithRetry create_customer_sharded.sql customer
+  sleep 4
+
+  echo "Apply 302_new_shards.yaml"
+  kubectl apply -f 302_new_shards.yaml
+  checkPodStatusWithTimeout "example-vttablet-zone1(.*)3/3(.*)Running(.*)" 8
+
+  killall kubectl
+  ./pf.sh > /dev/null 2>&1 &
+  sleep 5
+
+  waitForKeyspaceToBeServing customer -80 1
+  waitForKeyspaceToBeServing customer 80- 1
+
+  echo "Ready to reshard ..."
+  sleep 15
+
+  vtctlclient Reshard -source_shards '-' -target_shards '-80,80-' Create customer.cust2cust
+
+  sleep 5
+
+  vdiff_out=$(vtctlclient VDiff customer.cust2cust)
+  echo "$vdiff_out" | grep "ProcessedRows: 5" | wc -l | grep "2" > /dev/null
+  if [ $? -ne 0 ]; then
+    echo -e "VDiff output is invalid, got:\n$vdiff_out"
+    # Allow failure
+  fi
+
+  vtctlclient Reshard -tablet_types=rdonly,replica SwitchTraffic customer.cust2cust
+  vtctlclient Reshard -tablet_types=primary SwitchTraffic customer.cust2cust
+
+  sleep 10
+
+  assertSelect ../common/select_customer-80_data.sql "customer/-80" << EOF
+Using customer/-80
+Customer
++-------------+--------------------+
+| customer_id | email              |
++-------------+--------------------+
+|           1 | alice@domain.com   |
+|           2 | bob@domain.com     |
+|           3 | charlie@domain.com |
+|           5 | eve@domain.com     |
++-------------+--------------------+
+COrder
++----------+-------------+----------+-------+
+| order_id | customer_id | sku      | price |
++----------+-------------+----------+-------+
+|        1 |           1 | SKU-1001 |   100 |
+|        2 |           2 | SKU-1002 |    30 |
+|        3 |           3 | SKU-1002 |    30 |
+|        5 |           5 | SKU-1002 |    30 |
++----------+-------------+----------+-------+
+EOF
+
+  assertSelect ../common/select_customer80-_data.sql "customer/80-" << EOF
+Using customer/80-
+Customer
++-------------+----------------+
+| customer_id | email          |
++-------------+----------------+
+|           4 | dan@domain.com |
++-------------+----------------+
+COrder
++----------+-------------+----------+-------+
+| order_id | customer_id | sku      | price |
++----------+-------------+----------+-------+
+|        4 |           4 | SKU-1002 |    30 |
++----------+-------------+----------+-------+
+EOF
+
+  kubectl apply -f 306_down_shard_0.yaml
+  checkPodStatusWithTimeout "example-vttablet-zone1(.*)3/3(.*)Running(.*)" 6
+  waitForKeyspaceToBeServing customer -80 1
+  waitForKeyspaceToBeServing customer 80- 1
+}
+
+function upgradeToLatest() {
+  echo "Apply operator-latest.yaml "
+  kubectl apply -f operator-latest.yaml
+
+  echo "Upgrade all the other binaries"
+  kubectl apply -f cluster_upgrade.yaml
```

Comment:
nit: if you don't want to maintain two pairs of identic files: `operator-latest.yaml`-`operator.yaml`, and `101_initial_cluster.yaml`-`cluster_upgrade.yaml`, you can `sed` the files to change the tag of the Docker images

Reply:
Yes, I'll do that

Reply:
On second thought, or rather the reason I did it this way in the first place, was because we will be introducing a new flag very soon, and then upgrading to a new version of the operator will also require specifying that flag, which will also need to be added to the `operator-latest.yaml`. So, I didn't generate these files, since we would be throwing that code away in a separate PR real soon

Reply:
Got it, makes sense! What will that flag do?

Reply:
Our beloved, `durability_policy` is going to make its appearance here

Reply:
Gotcha!